src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala

```diff
@@ -114,23 +142,325 @@ class RequestHandler (configuration: Configuration) extends AppLogging {
+        case InstanceState.Running | InstanceState.NotReachable =>
+          instanceDao.setStateFor(instance.id.get, InstanceState.NotReachable)
+        case _ =>
+          instanceDao.setStateFor(instance.id.get, InstanceState.NotReachable)
+      }
+      log.info(s"Instance with id $id has reported stop.")
+      OperationResult.Ok
+    }
+  }
+
+  /** *
+    * Handles a call to /reportFailure. Needs the instance with the specified id to be present and running inside a docker
+    * container. Will update the state of the instance to 'Failed'. Will print warnings if illegal state transitions
+    * occur, but will update state anyway.
+    *
+    * @param id Id of the instance that reported failure
+    * @return OperationResult indicating either success or the reason for failure (which precondition was not met)
+    */
+  def handleReportFailure(id: Long, errorLog: Option[String]): OperationResult.Value = {
+    if (!instanceDao.hasInstance(id)) {
+      OperationResult.IdUnknown
+    } else if (!isInstanceDockerContainer(id)) {
+      OperationResult.NoDockerContainer
+    } else {
+      val instance = instanceDao.getInstance(id).get //TODO:Handle errorLog
+      instance.instanceState match {
+        case InstanceState.Failed =>
+          log.warning(s"Instance with id $id reported failure but state already was 'Failed'.")
+        case InstanceState.Paused =>
+          log.warning(s"Instance with id $id reported failure but state is 'Paused'. Will set state to 'Failed.")
+          instanceDao.setStateFor(instance.id.get, InstanceState.Failed)
+        case InstanceState.Stopped =>
+          log.warning(s"Instance with id $id reported failure but state is 'Stopped'. Will set state to 'Failed'.")
+          instanceDao.setStateFor(instance.id.get, InstanceState.Failed)
+        case _ =>
+          instanceDao.setStateFor(instance.id.get, InstanceState.Failed)
+      }
+      log.info(s"Instance with id $id has reported failure.")
+      OperationResult.Ok
+    }
+
+  }
+
+  /** *
+    * Handles a call to /pause. Needs instance with the specified id to be present, deployed inside a docker container,
+    * and running. Will pause the container and set state accordingly.
+    *
+    * @param id Id of the instance to pause
+    * @return OperationRsult indicating either success or the reason for failure (which preconditions failed)
+    */
+  def handlePause(id: Long): OperationResult.Value = {
+    if (!instanceDao.hasInstance(id)) {
+      OperationResult.IdUnknown
+    } else if (!isInstanceDockerContainer(id)) {
+      OperationResult.NoDockerContainer
+    } else {
+      val instance = instanceDao.getInstance(id).get
+      if (instance.instanceState == InstanceState.Running) {
+        log.info(s"Handling /pause for instance with id $id...")
+        //  dockerActor ! PauseMessage(instance.dockerId.get, instanceHasState())
+        //TODO: execute pause command (async?)
```

Comment:
Is is not possible to pause containers with the api we are using?

Reply:
I have implemented the pause functionality now.